src/opnsense/mvc/app/views/OPNsense/Firewall/filter_rule.volt

```diff
@@ -706,9 +726,15 @@
         $('#category_filter').parent().find('.dropdown-toggle').prepend('<i class="fa fa-tag" style="margin-right: 6px;"></i>');
 
         $("#reconfigureAct").SimpleActionButton({
-            onAction: function (data, status) {
-                populateInterfaceSelectpicker();
-                populateCategoriesSelectpicker();
+            onAction(data, status) {
+                reconfigureActInProgress = true;
+
+                Promise.all([
+                    new Promise(populateInterfaceSelectpicker),
+                    new Promise(populateCategoriesSelectpicker)
```

Comment:
Both functions are going to run sync here. If constructed using a Promise, you need to account for `resolve` (and optionally `reject`) in both functions as params. Technically you could even return a promise from `fetch_options`, but probably better to do it as is, and call `resolve()` from the completion callback